rotkehlchen/api/rest.py

```diff
@@ -5420,3 +5424,31 @@ def prepare_native_transfer(
             return wrap_in_fail_result(str(e), status_code=HTTPStatus.BAD_REQUEST)
 
         return _wrap_in_ok_result(result=payload)
+
+    @async_api_call()
+    def fetch_token_balance_for_address(
+            self,
+            address: ChecksumEvmAddress,
+            evm_chain: EVM_CHAIN_IDS_WITH_TRANSACTIONS_TYPE,
+            asset: 'CryptoAsset',
+    ) -> dict[str, Any]:
+        node_inquirer = self.rotkehlchen.chains_aggregator.get_evm_manager(evm_chain).node_inquirer
+        try:
+            if asset == node_inquirer.native_token:
+                balance = node_inquirer.get_multi_balance([address])[address]
+            elif (token := asset.resolve_to_evm_token()).chain_id == evm_chain:
+                balance = token_normalized_value(
+                    token_amount=node_inquirer.call_contract(
+                        contract_address=token.evm_address,
+                        abi=node_inquirer.contracts.erc20_abi,
+                        method_name='balanceOf',
+                        arguments=[address],
+                    ),
+                    token=token,
+                )
+            else:
+                raise InputError('Token exists on different chain than requested')
```

Comment:
why raise InputError? Isn't this the place to actually return directly? Never use exceptions for control flow like this